tempo_project/README.md

```diff
@@ -0,0 +1,398 @@
+```
+
+Note that although a column name is provided for each dimension type, trivial dimensions (i.e., those with only one possible value, like `weather_year`) are not included in the data files. Thus, not all column names listed in the metadata will actually be present in loaded data frames.
+
+
+#### examples-spark.ipynb
+
+Dependencies:
+- python=3.10
+- jupyter
+- pyspark
+- pandas
+- plotly
+
+Advantages: The data were originally created with Spark, and Spark can nominally work with all of the datasets.
+
+Limitations: Although set-up is easy for local mode, performing queries on large datasets generally requires a multi-node cluster, which can be challenging to set up and expensive to run.
+
+##### Getting Started
+
+PySpark easily loads parquet files even when those files are actually directories containing a partitioned dataset. For example, running this code:
+```
+from pyspark.sql import SparkSession
+
+spark = (
+            SparkSession.builder
+            .appName("dsgrid")
+            .config("spark.sql.session.timeZone", "EST")
+            .getOrCreate()
+        )
+
+dataset_name = "state_level_simplified"
+
+# Load data table
+filepath = data_dir / dataset_name / "table.parquet"
+df = spark.read.parquet(str(filepath))
+tablename = "tbl"
+df.createOrReplaceTempView(tablename)
+df.show(n=5)
+```
+in the notebook returns:
+![screenshot](docs/pyspark-load-data.png "Notebook output after loading 'state_level_simplified' into PySpark")
+
+##### Writing Queries
+
+[PySpark](https://spark.apache.org/docs/latest/api/python/index.html) data frames can be queried using functions like [filter](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.filter.html#pyspark.sql.DataFrame.filter) or by writing straight SQL. In the notebook we use SQL. For example, this query (when the data are small enough to process):
+```Python
+df = spark.sql(f"""SELECT scenario, {columns_by_type["model_year"]} as year, SUM({value_column})/1.0E6 as annual_twh
+                     FROM {tablename} 
+                 GROUP BY scenario, {columns_by_type["model_year"]}
+                 ORDER BY scenario, year""").toPandas()
+```
+returns a `pandas.DataFrame` containing three columns: `scenario`, `year`, and `annual_twh`.
+
+And this is an example of how to select timestamps within a range:
+```Python
+import datetime as dt
+    
+# Select EST timestamps for 2/14/2012
+start_timestamp = dt.datetime(2012, 2, 14, 0)
+stop_timestamp = dt.datetime(2012, 2, 14, 23)
+
+df = spark.sql(f"""SELECT time_est, SUM({value_column}) as {value_column}
+                     FROM {tablename} 
+                    WHERE {where_clause} AND (time_est >= TIMESTAMP '{start_timestamp}') AND (time_est <= TIMESTAMP '{end_timestamp}')
+                GROUP BY time_est 
+                ORDER BY time_est;""").toPandas()
+```
+
+##### Additional Reading
```

Comment:
@daniel-thom, are there specific documentation pages you would suggest for folks? (E.g., see what I listed under DuckDB Additional Reading.)